File: java/src/com/framgia/Part4/getuserinfo/UserTest.java
package com.framgia.Part4.getuserinfo;

import com.framgia.Part1.getuserinfo.GetUserInfoBetter;
import com.framgia.Part1.getuserinfo.User;
import org.junit.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.Assert.assertEquals;

public class UserTest {
    private GetUserInfoBetter getUserInfo = new GetUserInfoBetter();

    @Test
    public void test1() {
        List<User> users = new ArrayList<>();

        users.add(new User("Cong tan ton nu ta thi mong hang", "20-11-1991"));
        users.add(new User("Nguyen Van A2", "20-11-1992"));
        users.add(new User("Cong tan ton nu ta thi mong hang", "20-11-1991"));
        users.add(new User("Nguyen Van A2", "20-11-1992"));
        users.add(new User("Nguyen Van A3", "20-11-1993"));
        users.add(new User("Nguyen Van A4", "20-11-1994"));
        users.add(new User("Nguyen Van A5", "20-11-1995"));
        int expected = 125;
        assertEquals(expected, getUserInfo.calculateTotalAge(users));

        users.clear();
        users.add(new User("Cong tan ton nu ta thi mong hang", "20-11-1991"));
        users.add(new User("Nguyen Van A2", "20/11/1992"));
        users.add(new User("Nguyen Van A3", "20-11-1993"));
        users.add(new User("Nguyen Van A4", "20-11-1994"));
        users.add(new User("Nguyen Van A5", "20-11-1995"));
        expected = 125;
        assertEquals(expected, getUserInfo.calculateTotalAge(users));

        users.clear();
        users.add(new User("Cong tan ton nu ta thi mong hang", "20-11-1991"));
        users.add(new User("Nguyen Van A2", ""));
        users.add(new User("Nguyen Van A3", "20-11-1993"));
        users.add(new User("Nguyen Van A4", "20-11-1994"));
        users.add(new User("Nguyen Van A5", "20-11-1995"));
        expected = 109;
        assertEquals(expected, getUserInfo.calculateTotalAge(users));

        users.clear();
        users.add(new User("Cong tan ton nu ta thi mong hang", "20-11-1991"));
        users.add(new User("Nguyen Van A2", "20-11-1992"));
        users.add(new User("Nguyen Van A4", "20-11-1994"));
        users.add(new User("Nguyen Van A5", "20-11-1995"));
        expected = 100;
        assertEquals(expected, getUserInfo.calculateTotalAge(users));
    }
}

File: iOS/Part4/GetUserInfo/UserTest.swift
import XCTest

class UserTest: XCTestCase {
    
    private let getUserInfo = GetUserInfoBetter()
    
    override func setUp() {
        super.setUp()
        // Put setup code here. This method is called before the invocation of each test method in the class.
    }
    
    override func tearDown() {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
        super.tearDown()
    }
    
    func test1() {
        var users: [User] = []
    
        users.append(User(name: "Cong tan ton nu ta thi mong hang", birthday: "20-11-1991"))
        users.append(User(name: "Nguyen Van A2", birthday: "20-11-1992"))
        users.append(User(name: "Cong tan ton nu ta thi mong hang", birthday: "20-11-1991"))
        users.append(User(name: "Nguyen Van A2", birthday: "20-11-1992"))
        users.append(User(name: "Nguyen Van A3", birthday: "20-11-1993"))
        users.append(User(name: "Nguyen Van A4", birthday: "20-11-1994"))
        users.append(User(name: "Nguyen Van A5", birthday: "20-11-1995"))
        let expected = 125
        XCTAssertEqual(expected, getUserInfo.calculateTotalAge(users: users))
    
        users.removeAll()
        users.append(User(name: "Cong tan ton nu ta thi mong hang", birthday: "20-11-1991"))
        users.append(User(name: "Nguyen Van A2", birthday: "20/11/1992"))
        users.append(User(name: "Nguyen Van A3", birthday: "20-11-1993"))
        users.append(User(name: "Nguyen Van A4", birthday: "20-11-1994"))
        users.append(User(name: "Nguyen Van A5", birthday: "20-11-1995"))
        expected = 125
        XCTAssertEqual(expected, getUserInfo.calculateTotalAge(users: users))
    
        users.removeAll()
        users.append(User(name: "Cong tan ton nu ta thi mong hang", birthday: "20-11-1991"))
        users.append(User(name: "Nguyen Van A2", birthday: ""))
        users.append(User(name: "Nguyen Van A3", birthday: "20-11-1993"))
        users.append(User(name: "Nguyen Van A4", birthday: "20-11-1994"))
        users.append(User(name: "Nguyen Van A5", birthday: "20-11-1995"))
        expected = 109
        XCTAssertEqual(expected, getUserInfo.calculateTotalAge(users: users))
    
        users.removeAll()
        users.append(User(name: "Cong tan ton nu ta thi mong hang", birthday: "20-11-1991"))
        users.append(User(name: "Nguyen Van A2", birthday: "20-11-1992"))
        users.append(User(name: "Nguyen Van A4", birthday: "20-11-1994"))
        users.append(User(name: "Nguyen Van A5", birthday: "20-11-1995"))
        expected = 100
        XCTAssertEqual(expected, getUserInfo.calculateTotalAge(users: users))
    }
}
